torch/optim/adam.py

```diff
@@ -1,22 +1,34 @@
-from typing import List, Optional
+from typing import List, Optional, Union, Tuple
 
 import torch
 from torch import Tensor
-from .optimizer import (Optimizer, _use_grad_for_differentiable, _get_value, _stack_if_compiling,
-                        _dispatch_sqrt, _default_to_fused_or_foreach, _capturable_doc,
-                        _differentiable_doc, _foreach_doc, _fused_doc, _maximize_doc)
+from .optimizer import (Optimizer, params_t, _use_grad_for_differentiable, _get_value,
+                        _stack_if_compiling, _dispatch_sqrt, _default_to_fused_or_foreach,
+                        _capturable_doc, _differentiable_doc, _foreach_doc, _fused_doc,
+                        _maximize_doc)
 from torch.utils._foreach_utils import _get_fused_kernels_supported_devices
 
 __all__ = ['Adam', 'adam']
 
 
 class Adam(Optimizer):
-    def __init__(self, params, lr=1e-3, betas=(0.9, 0.999), eps=1e-8,
-                 weight_decay=0, amsgrad=False, *, foreach: Optional[bool] = None,
-                 maximize: bool = False, capturable: bool = False,
-                 differentiable: bool = False, fused: Optional[bool] = None):
+    def __init__(self,
+                 params: params_t,
+                 lr: Union[float, Tensor] = 1e-3,
```

Comment:
Does passing a single tensor here support multi device? shouldn't it be treated like the other scalars and be placed into an array?